File: src/frontend/wms/lib/app/data/model/wms_agendamento_model.dart
import 'dart:convert';
import 'package:pluto_grid/pluto_grid.dart';

import 'package:wms/app/infra/infra_imports.dart';
import 'package:intl/intl.dart';

class WmsAgendamentoModel {
	int? id;
	DateTime? dataOperacao;
	String? horaOperacao;
	String? localOperacao;
	int? quantidadeVolume;
	double? pesoTotalVolume;
	int? quantidadePessoa;
	int? quantidadeHora;

	WmsAgendamentoModel({
		this.id,
		this.dataOperacao,
		this.horaOperacao,
		this.localOperacao,
		this.quantidadeVolume,
		this.pesoTotalVolume,
		this.quantidadePessoa,
		this.quantidadeHora,
	});

	static List<String> dbColumns = <String>[
		'id',
		'data_operacao',
		'hora_operacao',
		'local_operacao',
		'quantidade_volume',
		'peso_total_volume',
		'quantidade_pessoa',
		'quantidade_hora',
	];
	
	static List<String> aliasColumns = <String>[
		'Id',
		'Data Operacao',
		'Hora Operacao',
		'Local Operacao',
		'Quantidade Volume',
		'Peso Total Volume',
		'Quantidade Pessoa',
		'Quantidade Hora',
	];

	WmsAgendamentoModel.fromJson(Map<String, dynamic> jsonData) {
		id = jsonData['id'];
		dataOperacao = jsonData['dataOperacao'] != null ? DateTime.tryParse(jsonData['dataOperacao']) : null;
		horaOperacao = jsonData['horaOperacao'];
		localOperacao = jsonData['localOperacao'];
		quantidadeVolume = jsonData['quantidadeVolume'];
		pesoTotalVolume = jsonData['pesoTotalVolume']?.toDouble();
		quantidadePessoa = jsonData['quantidadePessoa'];
		quantidadeHora = jsonData['quantidadeHora'];
	}

	Map<String, dynamic> get toJson {
		Map<String, dynamic> jsonData = <String, dynamic>{};

		jsonData['id'] = id != 0 ? id : null;
		jsonData['dataOperacao'] = dataOperacao != null ? DateFormat('yyyy-MM-ddT00:00:00').format(dataOperacao!) : null;
		jsonData['horaOperacao'] = Util.removeMask(horaOperacao);
		jsonData['localOperacao'] = localOperacao;
		jsonData['quantidadeVolume'] = quantidadeVolume;
		jsonData['pesoTotalVolume'] = pesoTotalVolume;
		jsonData['quantidadePessoa'] = quantidadePessoa;
		jsonData['quantidadeHora'] = quantidadeHora;
	
		return jsonData;
	}
	
	String objectEncodeJson() {
		final jsonData = toJson;
		return json.encode(jsonData);
	}

	plutoRowToObject(PlutoRow plutoRow) {
		id = plutoRow.cells['id']?.value;
		dataOperacao = Util.stringToDate(plutoRow.cells['dataOperacao']?.value);
		horaOperacao = plutoRow.cells['horaOperacao']?.value;
		localOperacao = plutoRow.cells['localOperacao']?.value;
		quantidadeVolume = plutoRow.cells['quantidadeVolume']?.value;
		pesoTotalVolume = plutoRow.cells['pesoTotalVolume']?.value?.toDouble();
		quantidadePessoa = plutoRow.cells['quantidadePessoa']?.value;
		quantidadeHora = plutoRow.cells['quantidadeHora']?.value;
	}	

	WmsAgendamentoModel clone() {
		return WmsAgendamentoModel(
			id: id,
			dataOperacao: dataOperacao,
			horaOperacao: horaOperacao,
			localOperacao: localOperacao,
			quantidadeVolume: quantidadeVolume,
			pesoTotalVolume: pesoTotalVolume,
			quantidadePessoa: quantidadePessoa,
			quantidadeHora: quantidadeHora,
		);			
	}

	
}

File: backend/wms/src/model/wms_agendamento_model.py
from src import db


class WmsAgendamentoModel(db.Model):
    __tablename__ = 'wms_agendamento'

    id = db.Column(db.Integer, primary_key=True)
    data_operacao = db.Column(db.DateTime)
    hora_operacao = db.Column(db.String(8))
    local_operacao = db.Column(db.String(100))
    quantidade_volume = db.Column(db.Integer)
    peso_total_volume = db.Column(db.Float)
    quantidade_pessoa = db.Column(db.Integer)
    quantidade_hora = db.Column(db.Integer)


    def mapping(self, data):
        self.id = data.get('id')
        self.data_operacao = data.get('dataOperacao')
        self.hora_operacao = data.get('horaOperacao')
        self.local_operacao = data.get('localOperacao')
        self.quantidade_volume = data.get('quantidadeVolume')
        self.peso_total_volume = data.get('pesoTotalVolume')
        self.quantidade_pessoa = data.get('quantidadePessoa')
        self.quantidade_hora = data.get('quantidadeHora')

    def serialize(self):
        return {
            'id': self.id,
            'dataOperacao': self.data_operacao.isoformat(),
            'horaOperacao': self.hora_operacao,
            'localOperacao': self.local_operacao,
            'quantidadeVolume': self.quantidade_volume,
            'pesoTotalVolume': self.peso_total_volume,
            'quantidadePessoa': self.quantidade_pessoa,
            'quantidadeHora': self.quantidade_hora,
        }